### PR TITLE
Adds support for softmax

### PR DESCRIPTION
**Pull request by mxmasterton**
Implements exp and log operations as well as softmax

### CODE REVIEW

**Review by nadeemcv — APPROVED** [review]

**Review by 54g0 — CHANGES_REQUESTED** [review]
It is wrong the backward function inside the log function is wrong because the derivative of logx is 1/x so self.grad += (1/self.data)*out.grad
```python 

def log(self):
    out = Value(math.log(self.data),(self,),"log")
    def _backward():
        self.grad += (1/self.data)* out.grad
    out._backward = _backward
    return out 
```
and one more bug you by default python start counting from the integer 0 so it tries to do 0 + Value() so Value class doest know how to add a plain integer from the left side so this would fail the correct way is denom = sum(exps,Value(0.0))
```python
@staticmethod
    def softmax(logits):
        exps = [logit.exp() for logit in logits]
        denom = sum(exps,Value(0.0))
        return [e/denom for e in exps]
```

**Comment by mxmasterton** [comment]
Will fix the first bug in a second. Second one is not an issue because radd is implemented and also there is a check right at the start of add to catch this.
